src/app/[locale]/client-search-dataset.ts

```diff
@@ -0,0 +1,29 @@
+import {SearchResult} from '@/lib/dataset-fetcher';
+
+interface SearchDatasets {
+  licenses: string[];
+  publishers: string[];
+}
+
+// Only use this function for client components.
+// For server component use the dataset-fetcher directly.
+export async function clientSearchDatasets({
+  licenses,
+  publishers,
+}: SearchDatasets): Promise<SearchResult> {
+  const searchParams = {
+    licenses: licenses.join(','),
+    publishers: publishers.join(','),
```

Comment:
These transformations are identical - perhaps move to an inline function, to have the logic in one place? This also makes it easier to add more logic if required (e.g. we may not want to set the licenses and publishers if the result of the join() is an empty string?).

Reply:
I'm used to working with [the rule of 3](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)). This rule states that two instances of similar code do not require refactoring, but when a similar code is used three times, it should be extracted into a new procedure.

Reply:
Nice! I'm used to working with a rule of 2 ('two or more, use a for'). But I'm happy as long as we have a philosophy about this in mind :)